app/scripts/views/elements/unblock-code-input.js

```diff
@@ -0,0 +1,31 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+define(function (require, exports, module) {
+  'use strict';
+
+  const AuthErrors = require('lib/auth-errors');
+  const Constants = require('lib/constants');
+  const textInput = require('views/elements/text-input');
+
+  const element = Object.create(textInput);
+
+  element.match = function ($el) {
+    return $el.attr('type') === 'text' && $el.hasClass('unblock-code');
+  };
+
+  element.validate = function () {
+    const isRequired = typeof this.attr('required') !== 'undefined';
```

Comment:
this seems like a weird check, especially if we are using jquery. might be able to simplify